test/framework/src/main/java/org/elasticsearch/cluster/service/ClusterStateTaskExecutorUtils.java

```diff
@@ -82,6 +82,38 @@ public static <T extends ClusterStateTaskListener> ClusterState executeHandlingR
         return resultingState;
     }
 
+    public static <T extends ClusterStateTaskListener> ClusterState executeExecutingResults(
+        ClusterState originalState,
+        ClusterStateTaskExecutor<T> executor,
+        Iterable<T> tasks,
+        CheckedConsumer<T, Exception> onTaskSuccess,
+        CheckedBiConsumer<T, Exception, Exception> onTaskFailure
+    ) throws Exception {
+        final var taskContexts = StreamSupport.stream(tasks.spliterator(), false).<ClusterStateTaskExecutor.TaskContext<T>>map(
+            t -> new TestTaskContext<>(t) {
+                @Override
+                public void success(Runnable onPublishSuccess) {
+                    super.success(onPublishSuccess);
+                    onPublishSuccess.run();
```

Comment:
@DaveCTurner I had to add this method to explicitly call the runnable that is passed in, so the jointask listener actually gets called. Is there a reason this isn't done in the first place?

Reply:
Hm not sure, it does seem strange not to invoke this in the other utils too. Does anything break if we do that?